File: codegen/results/v002/m3-max-48gb/ollama/qwen3.5_27b_nothink/20260411_194638/python/004_json_filter/solution.py
import json
import sys

def main():
    input_file = "input/data.json"
    
    try:
        with open(input_file, 'r', encoding='utf-8') as f:
            data = json.load(f)
    except FileNotFoundError:
        sys.exit(1)
    except json.JSONDecodeError:
        sys.exit(1)

    # Filter records where active is True and age >= 30
    filtered = [
        record for record in data
        if record.get("active") is True and record.get("age", 0) >= 30
    ]

    # Sort by name ascending
    filtered.sort(key=lambda x: x.get("name", ""))

    # Output the result as a JSON array to stdout
    print(json.dumps(filtered))

if __name__ == "__main__":
    main()

File: codegen/results/v002/m3-max-48gb/ollama/qwen3.5_27b_nothink/20260411_194638/typescript/004_json_filter/solution.ts
import * as fs from 'fs';
import * as path from 'path';

// Define the type for the record
interface Person {
  name: string;
  age: number;
  active: boolean;
  score: number;
}

function main() {
  // Resolve the path to the input file
  const filePath = path.join(__dirname, 'input', 'data.json');

  // Read the file content
  const rawData = fs.readFileSync(filePath, 'utf-8');

  // Parse the JSON data
  const data: Person[] = JSON.parse(rawData);

  // Filter: active must be true AND age must be 30 or older
  const filtered = data.filter(
    (person) => person.active === true && person.age >= 30
  );

  // Sort: by name ascending
  const sorted = filtered.sort((a, b) => {
    if (a.name < b.name) return -1;
    if (a.name > b.name) return 1;
    return 0;
  });

  // Output the result as a JSON string to stdout
  console.log(JSON.stringify(sorted));
}

main();